pkg/settings/manager.go

```diff
@@ -0,0 +1,122 @@
+package settings
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"time"
+
+	bolt "go.etcd.io/bbolt"
+)
+
+// Settings represents user settings.
+type Settings struct {
+    // OpenaiKey is an OpenAI API Key to use with Assistant for GraphQL queries.
+	OpenaiKey string `json:"openai_key"`
+
+	// CreatedAt is the time the settings were created.
+    CreatedAt time.Time `json:"created_at"`
+
+    // UpdatedAt is the time the settings were updated.
+    UpdatedAt time.Time `json:"updated_at"`
+}
+
+type Manager struct {
+	db *bolt.DB
+}
+
+// NewManager creates a new settings manager.
+func NewManager(db *bolt.DB) *Manager {
+	return &Manager{db: db}
+}
+
+// UpdateSettings updates the fields of a Settings instance in the database.
+func (m *Manager) UpdateSettings(ctx context.Context, settings *Settings) error {
```

Comment:
I think the code might be a bit overcomplicated given that we’re working with a key-value store like bbolt. 

Instead of using reflection to iterate over the fields and preserve `CreatedAt`, we could simplify this by assuming that the new settings object will have all the necessary fields so we can directly marshal and store it.


There is no need to have a created_at field for settings. Feel free to delete the Field in the struct + simplify this.

Reply:
Do you mean making it as simple as this? 

// UpdateSettings updates the fields of a Settings instance in the database.
func (m *Manager) UpdateSettings(ctx context.Context, settings *Settings) error {
	return m.db.Update(func(tx *bolt.Tx) error {
		b, err := tx.CreateBucketIfNotExists([]byte("settings"))
		if err != nil {
			return fmt.Errorf("create bucket: %w", err)
		}

		settings.UpdatedAt = time.Now().UTC()
        
       // Marshal the settings back to JSON
       value, err = json.Marshal(settings)
       if err != nil {
        return fmt.Errorf("encode settings: %w", err)
       }

		// Put the settings back into the bucket
		return b.Put(key, value)
	})
}

Reply:
Exactly